pgjdbc/src/main/java/org/postgresql/core/TypeInfo.java

```diff
@@ -91,6 +92,8 @@ void addCoreType(String pgTypeName, Integer oid, Integer sqlType, String javaCla
 
   @Nullable Class<? extends PGobject> getPGobject(String type);
 
+  @Nullable PgStruct getPgStruct(String type) throws SQLException;
```

Comment:
It seems like maybe we need a different class to be a "struct descriptor" that would be in TypeInfo.
then PgStruct would implement the jdbc Struct spec and represent an actual instance.